File: solutions/3.cpp
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        int maxLen = 0;
        unordered_set<char> set;
        int left = 0;
        for (int r = 0; r < s.length(); r++)
        {
            while (set.count(s[r]))
            {
                set.erase(s[left]);
                left++;
            }
            set.insert(s[r]);
            maxLen = max(maxLen, r - left + 1);
        }
        return maxLen;
    }
};

File: solutions/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = 0
        left = 0
        ch_map = set()
        for right in range(len(s)):
            while s[right] in ch_map:
                ch_map.remove(s[left])
                left += 1
            ch_map.add(s[right])
            longest = max(longest, right - left + 1)
        return longest